### PR TITLE
Add convenient methods on branch/injection for PQI and connection.

### LINKED ISSUE #1536: [IIDM] P, Q, I availability on Injection and Branch classes
* **Do you want to request a *feature* or report a *bug*?**

Feature.

* **What is the current behavior?**

Some frequently used physical quantities like power flows and currents are only available on `Terminal` objects, whereas one could expect to have them available on the network equipments themselves, more specifically on `Branch` and on `Injection` classes.

For example to get the current on side 1 of a line, we need to write :
```java
double p1 = line.getTerminal1().getP();
```
or for a generator : 
```java
double p = generator.getTerminal().getP();
//surprising because on the other side we can retrieve targetP directly:
double targetP = generator.getTargetP();
```

* **What is the expected behavior?**

The user should be able to retrieve those quantities more simply:
```java
double p1 = branch.getP1();
double p2 = branch.getP2();
double p = generator.getP();
```
Such shortcuts should be available on `Injection` and `Branch` for:
 * P (getter/setter)
 * Q (getter/setter)
 * I
 * connect/disconnect
 * voltage level

* **What is the motivation / use case for changing the behavior?**

Making the API a little more user-friendly.

### PR DESCRIPTION
**Pull request by yichen88**
Currently we have to bypass equipment's terminal to modify or access those physical quantities.
In this implements, we add default methods on interface to access those variables.

Signed-off-by: yichen88 <tang.yichenyves@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

fix #1536 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java [hunk 1]
```diff
@@ -179,4 +179,162 @@
...
+        return getTerminal1().getP();
+    }
+
+    /**
+     * Set the active power in MW injected at side one of this equipment.
+     * <p>
+     * Depends on the working variant.
+     * @see VariantManager
+     */
+    default Branch<I> setP1(double p) {
```
Comment (MioRtia) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java:199]: For me, it should return directly `I` not `Branch<I>` (same remark for other setters)

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Injection.java [hunk 2]
```diff
@@ -18,4 +18,83 @@
...
+        return getTerminal().getP();
+    }
+
+    /**
+     * Set the active power in MW injected at this equipment.
+     * <p>
+     * Depends on the working variant.
+     * @see VariantManager
+     */
+    default Injection<I> setP(double p) {
```
Comment (MioRtia) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Injection.java:37]: Same remark as above, should return `I`

**Review by MioRtia — COMMENTED** [review]
Not sure if you should put the implementation in the implementation classes as well (even if it is the same as the default implementation)?

File: iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTest.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+public class BranchTest {
+
+    @Test
+    public void testPQI() {
+        Terminal t1 = mock(Terminal.class);
+        Terminal t2 = mock(Terminal.class);
+        Branch branch = new Branch() {
```
Comment (sylvlecl) [inline iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTest.java:29]: Might be shorter to use a Mock + `thenCallRealMethod()`. Not sure it really has some added value to be honest ...

**Review by sylvlecl — APPROVED** [review]

**Review by MioRtia — APPROVED** [review]
I agree with Sylvain's comment, otherwise this is great for me!

**Pull request closed without merging**